modules/dockermodelrunner/internal/sdk/client/pull.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+)
+
+// pullOptions contains options for pulling a model
+type pullOptions struct {
+	// Progress is an optional writer to show download progress
+	Progress ProgressWriter
+}
+
+// PullOption is a function that configures PullOptions
+type PullOption func(*pullOptions)
+
+// PullModel creates a model in the Docker Model Runner, by pulling the model from Docker Hub.
+func (c *Client) PullModel(ctx context.Context, fqmn string, opts ...PullOption) error {
```

Comment:
suggestion: I'm guessing this is fully qualified model name, but it's not 100% obvious so I would use a more obvious name even if it is longer.